### ISSUE TITLE
[TTNN Dialect] Convert to_memory_config from DPS to non-DPS op

### ISSUE THREAD

**Issue by svuckovicTT** [tenstorrent/tt-mlir#436]
The op should be converted to match actual implementation in ttnn library.

**Comment by tapspatel** [comment]
Reopening issue. I had to revert this commit since it broke main for silicon tests with the following error signature

```
Traceback (most recent call last):
  File "/opt/ttmlir-toolchain/venv/bin/ttrt", line 8, in <module>
    sys.exit(main())
  File "/opt/ttmlir-toolchain/venv/lib/python3.10/site-packages/ttrt/__init__.py", line 42, in main
    request_api()
  File "/opt/ttmlir-toolchain/venv/lib/python3.10/site-packages/ttrt/common/api.py", line 940, in __call__
    self.execute()
  File "/opt/ttmlir-toolchain/venv/lib/python3.10/site-packages/ttrt/common/api.py", line 905, in execute
    _execute(self.ttnn_binaries)
  File "/opt/ttmlir-toolchain/venv/lib/python3.10/site-packages/ttrt/common/api.py", line 864, in _execute
    event = ttrt.runtime.submit(
IndexError: _Map_base::at
```

Please see ttrt docs on how to run test cases for n150 + n300 systems. We are working in the background on getting CI up and running for this.

fyi @jnie-TT

**Comment by jnie-TT** [comment]
@svuckovicTT I took a look at this. It seems like the final toMemoryConfig op that moves the output to host is getting an invalid input_id that does not exist in the live_tensors map in runtime. That's why we are getting a map::base at error.

I narrowed it down to this:
```
  auto input = cache.getOrCreate(op.getInput(), tensorValueToFlatbuffer,
                                 kHostAllocatedAddress, kHostAllocatedSize);
```
in TTNNToFlatbuffer.cpp. getOrCreate here should be able to retrieve the input from the flatbuffer cache (this getInput() should reference the output of the previous op). However, for some reason it cannot find the input and creates a new one instead. This new input is completely detached from the execution graph because it's not linked to the output of any previous operation. 

For example, looking at this MLIR snippet: 
```
    %0 = "ttnn.open_device"() <{device_ids = [0, 1]}> : () -> !tt.device<<#tt.grid<7x8, (d0, d1) -> (0, d0, d1)>, [0]>>
    %1 = "ttnn.to_memory_config"(%arg0) : (tensor<64x128xf32, #layout>) -> tensor<64x128xf32, #layout2>
    %2 = "ttnn.empty"(%0) : (!tt.device<<#tt.grid<7x8, (d0, d1) -> (0, d0, d1)>, [0]>>) -> tensor<64x128xf32, #layout2>
    %3 = "ttnn.relu"(%1, %2) <{operandSegmentSizes = array<i32: 1, 1>}> : (tensor<64x128xf32, #layout2>, tensor<64x128xf32, #layout2>) -> tensor<64x128xf32, #layout2>
    %4 = "ttnn.to_memory_config"(%3) : (tensor<64x128xf32, #layout2>) -> tensor<64x128xf32, #layout1>
    "ttnn.close_device"(%0) : (!tt.device<<#tt.grid<7x8, (d0, d1) -> (0, d0, d1)>, [0]>>) -> ()
    return %4 : tensor<64x128xf32, #layout1>
```
%3 inputted into the to_memory_config operation should be the output of the ttnn.relu op. However, `op.getInput()` returns a value that's different from the output of the relu op, and thus a new RankedTensor is created with a globa_id that doesn't previously exist

**Comment by svuckovicTT** [comment]
Ah, that's too bad. There's some discussion on that topic here: https://github.com/tenstorrent/tt-mlir/issues/435

I'll give it a think today.

**Comment by svuckovicTT** [comment]
Fixed via #437.
